Sources/Core/SourceFile.swift

```diff
@@ -33,6 +33,14 @@ public struct SourceFile {
     try self.init(contentsOf: URL(fileURLWithPath: filePath))
   }
 
+  /// Creates an instance representing the in-memory edited file at `filePath`.
+  /// `withContent` is the content of the in-memory representation of the file,
+  /// which may be different from what is actually stored on disk.
+  public init(at filePath: URL, withContent content: String) {
```

Comment:
```suggestion
  /// Creates an instance representing the in-memory `contents` of the file at `filePath`.
  ///
  /// `contents` is the text of the file currently in memory, which may be different from what
  /// is actually stored on disk when the file is being edited.
  public init(at filePath: URL, withInMemoryContents contents: String) {
```

Note: `contents` with a "s" is consistent with how other parameters are named.